### PR TITLE
Add library card support to switch_db_hash command.

### PR DESCRIPTION
**Pull request by demiankatz**
The command for changing database encryption did not account for library cards.

TODO
- [x] Improve test coverage
- [x] Update changelog when merging (changed constructor signature in command)

### CODE REVIEW

File: module/VuFindConsole/src/VuFindConsole/Command/Util/SwitchDbHashCommand.php [hunk 1]
```diff
@@ -145,6 +159,26 @@ protected function getOpenSsl($algorithm)
...
+     * @param UserRow|UserCardRow $row       Row to update
+     * @param ?BlockCipher        $oldcipher Old cipher (null for none)
+     * @param BlockCipher         $newcipher New cipher
+     *
+     * @return void
+     * @throws InvalidArgumentException
+     */
+    protected function fixRow($row, ?BlockCipher $oldcipher, BlockCipher $newcipher): void
+    {
+        $pass = ($oldcipher && $row['cat_pass_enc'] ?? null !== null)
```
Comment (EreMaijala) [inline module/VuFindConsole/src/VuFindConsole/Command/Util/SwitchDbHashCommand.php]: `&&` has a higher precedence than `??` so needs parentheses around `$row['cat_pass_enc'] ?? null`, right?

Comment (EreMaijala) [reply module/VuFindConsole/src/VuFindConsole/Command/Util/SwitchDbHashCommand.php]: Or do we need null coalescing at all? The column always exists even if null.

**Review by EreMaijala — CHANGES_REQUESTED** [review]

File: module/VuFindConsole/src/VuFindConsole/Command/Util/SwitchDbHashCommand.php (hunk 1 above)
Comment (demiankatz) [reply module/VuFindConsole/src/VuFindConsole/Command/Util/SwitchDbHashCommand.php]: Both good points -- null coalescing removed. That operator precedence thing is a common trap!

**Review by EreMaijala — APPROVED** [review]

File: module/VuFindConsole/src/VuFindConsole/Command/Util/SwitchDbHashCommand.php (hunk 1 above)
Comment (EreMaijala) [reply module/VuFindConsole/src/VuFindConsole/Command/Util/SwitchDbHashCommand.php]: It really is!

**Pull request merged**